lib/rules/semi.js

```diff
@@ -53,7 +53,7 @@ module.exports = {
 
     create(context) {
 
-        const OPT_OUT_PATTERN = /[\[\(\/\+\-]/; // One of [(/+-
+        const OPT_OUT_PATTERN = /[\[\(\/](\+[^\+]|\-[^\-])/; // One of [(/+-, but not ++ or --
```

Comment:
Thanks, I simplified the regular expression, fixed it for your example and added it as a test.